Add Support for Geekworm MZP280 in vc4-dpi-panel-overlay

macromorgan opened the pull request:
This pull request is to add support for the mzp280 panel in the generic vc4-dpi-panel-overlay that was merged yesterday. Note that since this panel is rotated 270 (depending upon your application anyway) rotation support is also added to the vc4-dpi-panel overlay.

6by9 commented on arch/arm/boot/dts/overlays/README:
```diff
@@ -3683,6 +3683,7 @@ Params: at056tn53v1             Enable an Innolux 5.6in VGA TFT
         backlight-def-brightness
                                 Set the default brightness. Normal range 1-16.
                                 (default 16).
+        rotate                  Display rotation {0,90,180,270} (default 0)
```
This also needs to go in the README for vc4-kms-dpi-generic and vc4-kms-kippah-7inch because they also include vc4-kms-dpi.dtsi.

Otherwise looks fine to me.

macromorgan replied:
vc4-kms-kippah-7inch seems to be depreciated in favor of vc4-kms-dpi-panel. I'll update vc4-kms-dpi-generic and resubmit though, thank you.

macromorgan replied:
Nevermind, I understand. Even though it's depreciated it's still a valid devicetree (for now). I'll update it too, thank you.

macromorgan replied:
Fixed and force pushed a new update.

pelwell replied:
You don't need to add it for vc4-kms-kippah-7inch - the deprecation (added very recently so 6by9 probably hadn't seen it) means the parameters aren't listed.

6by9 replied:
I'd seen your firmware mod to support renaming to a new overlay with params, but hadn't seen you'd updated the README. Sorry, my bad.

macromorgan replied:
Okay, so to confirm I need to 1) add it to the vc4-kms-dpi-generic, but not to the vc4-kms-kippah-7inch, right?

Also, it's not a big deal but I see a warning about using a dummy regulator. Should I just add for this specific device a power-supply (would point it to whatever the main 5v one is, since that what this runs off of)?

6by9 replied:
Correct. Add to vc4-kms-dpi-generic, but not to the vc4-kms-kippah-7inch.

Messages about dummy regulators are almost inevitable. You can create a regulator with ```compatible = "regulator-fixed";``` if you really felt like it, but it's not really necessary.

macromorgan replied:
I won't worry about fixed regulators then for now (I tried it and there's no gain really aside from one less thing in dmesg).

I just updated the pull with the README fixes. Let me know if there is anything else that's needed. Thank you.

macromorgan committed:
overlays: Add rotate property to vc4-kms-dpi-panel

Allow a user to specify the panel rotation in devicetree as 0, 90,
180, or 270 by setting a parameter.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>

macromorgan committed:
overlays: Add Geekworm mzp280 to vc4-kms-dpi-panel

Add support for the Geekworm mzp280 DPI panel to the generic
vc4-kms-dpi-panel overlay.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
